Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -435,8 +435,33 @@ namespace AZ::DocumentPropertyEditor
                     instancePointerValue,
                     access.Get(),
                     attributes,
-                    [](const Dom::Value& newValue)
+                    // this needs to write the value back into the reflected object via Json serialization
+                    [valuePointer = access.Get(), valueType = access.GetType(), this](const Dom::Value& newValue)
                     {
+                        // marshal this new value into a pointer for use by the Json serializer
+                        auto marshalledPointer = AZ::Dom::Utils::TryMarshalValueToPointer(newValue, valueType);
+
+                        rapidjson::Document buffer;
+                        JsonSerializerSettings serializeSettings;
+                        JsonDeserializerSettings deserializeSettings;
+                        serializeSettings.m_serializeContext = m_serializeContext;
+                        deserializeSettings.m_serializeContext = m_serializeContext;
+
+                        // serialize the new value to Json, using the original valuePointer as a reference object to generate a minimal diff
+                        JsonSerialization::Store(buffer, buffer.GetAllocator(), marshalledPointer, valuePointer, valueType, serializeSettings);
+
+                        // now deserialize that value into the original location
+                        JsonSerialization::Load(valuePointer, valueType, buffer, deserializeSettings);
+
+                        // NB: the returned value for serialized pointer values is instancePointerValue, but since this is passed by pointer,
+                        // it will not actually detect a changed dom value. Since we are already writing directly to the DOM before this step,
+                        // it won't affect the calling DPE, however, other DPEs pointed at the same adapter would be unaware of the change,
+                        // and wouldn't update their UI.
+                        // In future, to properly support multiple DPEs on one adapter, we will need to solve this. One way would be to store
+                        // the json serialized value (which is mostly human-readable text) as an attribute, so any change to the Json would
+                        // trigger an update. This would have the advantage of allowing opaque and pointer types to be searchable by the
+                        // string-based Filter adapter. Without this, things like Vector3 will not have searchable values by text. These
+                        // advantages would have to be measured against the size changes in the DOM and the time taken to populate and parse them.
```

Comment:
One of the arduant discussion we had about allowing the opaque pointer was that it was not able to be persistent to disk and that it would only be used for in-memory operations.
> Opaque: An arbitrary value stored in an AZStd::any. This is a non-serializable representation of an entry used only for in-memory options. This is intended to be used as an intermediate value over the course of DOM transformation and as a proxy to pass through types of which the DOM has no knowledge to other systems.